Position not set error for `SourceFile` annotation, when accessed from a `transparent` macro

## Compiler version

3.2.1

## Minimized code

The following repository contains a minimized reproducer: https://github.com/vigoo/scala3-annotation-bug

Below I'm summarizing it but for the crash it is important to have two separate compiled modules.

Let's define a custom annotation and a case class that uses it:

```scala
import scala.annotation.StaticAnnotation

final case class testAnnotation(steps: String) extends StaticAnnotation

@testAnnotation("x")
final case class Record(a: String, b: Int)
```

then we create a macro that tries to collect all annotations of a given type:

```scala
case class Schema(annotations: List[Any])

object DeriveSchema {
  transparent inline def gen[T]: Schema = ${ deriveSchema[T] }

  def deriveSchema[T: Type](using ctx: Quotes): Expr[Schema] = {
    import ctx.reflect._

    val typeRepr = TypeRepr.of[T]
    val annotationExprs = TypeRepr.of[T].typeSymbol.annotations.map(_.asExpr)
//    val annotations = '{List.empty}
    val annotations = '{List(${Expr.ofSeq(annotationExprs)})} // NOTE: if this line is just an empty list it does not crash

    '{Schema($annotations)}
  }
}
```

and call it:

```scala
@main def run() = {
  val schema: Schema = DeriveSchema.gen[Record]
  println(schema)
}
```

if `Record` is defined in a different compilation unit (a separate sbt subproject in the linked repo) the compiler crashes. If `Record` is defined next to the macro invocation it works. Also if the `val annotations = ` line is replaced to the commented out one, it does not crash.

## Output (click arrow to expand)
<details>

```
[info] compiling 1 Scala source to /home/vigoo/projects/oss/scala3-annotation-bug/base/target/scala-3.2.1/classes ...
[info] compiling 2 Scala sources to /home/vigoo/projects/oss/scala3-annotation-bug/dep/target/scala-3.2.1/classes ...
java.lang.AssertionError: assertion failed: position not set for new scala.annotation.internal.SourceFile("base/src/main/scala/demo/Base.scala") # -1 of class dotty.tools.dotc.ast.Trees$Apply in <no file> while typechecking /home/vigoo/projects/oss/scala3-annotation-bug/dep/src/main/scala/demo/Dep.scala
[info] exception occurred while typechecking /home/vigoo/projects/oss/scala3-annotation-bug/dep/src/main/scala/demo/Dep.scala
java.lang.AssertionError: assertion failed: position not set for new scala.annotation.internal.SourceFile("base/src/main/scala/demo/Base.scala") # -1 of class dotty.tools.dotc.ast.Trees$Apply in <no file> while compiling /home/vigoo/projects/oss/scala3-annotation-bug/dep/src/main/scala/demo/Dep.scala, /home/vigoo/projects/oss/scala3-annotation-bug/dep/src/main/scala/demo/macros.scala
[error] ## Exception when compiling 2 sources to /home/vigoo/projects/oss/scala3-annotation-bug/dep/target/scala-3.2.1/classes
[error] java.lang.AssertionError: assertion failed: position not set for new scala.annotation.internal.SourceFile("base/src/main/scala/demo/Base.scala") # -1 of class dotty.tools.dotc.ast.Trees$Apply in <no file>
[error] scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
[error] dotty.tools.dotc.typer.Typer$.assertPositioned(Typer.scala:71)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3045)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error] dotty.tools.dotc.typer.Typer.$anonfun$36(Typer.scala:1898)
[error] dotty.tools.dotc.core.Decorators$ListDecorator$.loop$1(Decorators.scala:87)
[error] dotty.tools.dotc.core.Decorators$ListDecorator$.mapconserve$extension(Decorators.scala:103)
[error] dotty.tools.dotc.typer.Typer.typedSeqLiteral(Typer.scala:1898)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2941)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error] dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error] dotty.tools.dotc.typer.Typer.typedInlined(Typer.scala:1917)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2942)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error] dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error] dotty.tools.dotc.typer.Typer.$anonfun$36(Typer.scala:1898)
[error] dotty.tools.dotc.core.Decorators$ListDecorator$.loop$1(Decorators.scala:87)
[error] dotty.tools.dotc.core.Decorators$ListDecorator$.mapconserve$extension(Decorators.scala:103)
[error] dotty.tools.dotc.typer.Typer.typedSeqLiteral(Typer.scala:1898)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2941)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error] dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error] dotty.tools.dotc.typer.ReTyper.typedTyped(ReTyper.scala:64)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2925)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error] dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.$anonfun$7(ProtoTypes.scala:483)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.cacheTypedArg(ProtoTypes.scala:406)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:484)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:866)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:866)
[error] dotty.tools.dotc.typer.Applications$Application.addTyped$1(Applications.scala:557)
[error] dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:603)
[error] dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:460)
[error] dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:748)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:865)
[error] dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1077)
[error] dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:327)
[error] dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:121)
[error] dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:923)
[error] dotty.tools.dotc.typer.Applications.realApply$1$$anonfun$3(Applications.scala:1003)
[error] dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:3194)
[error] dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:1014)
[error] dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1052)
[error] dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:327)
[error] dotty.tools.dotc.inlines.Inliner$InlineTyper.typedApply(Inliner.scala:812)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2920)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error] dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error] dotty.tools.dotc.typer.Typer.typedInlined(Typer.scala:1917)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2942)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error] dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error] dotty.tools.dotc.typer.Typer.typedInlined(Typer.scala:1917)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2942)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error] dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.$anonfun$7(ProtoTypes.scala:483)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.cacheTypedArg(ProtoTypes.scala:406)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:484)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:866)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:866)
[error] dotty.tools.dotc.typer.Applications$Application.addTyped$1(Applications.scala:557)
[error] dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:622)
[error] dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:460)
[error] dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:748)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:865)
[error] dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1077)
[error] dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:327)
[error] dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:121)
[error] dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:923)
[error] dotty.tools.dotc.typer.Applications.realApply$1$$anonfun$3(Applications.scala:1003)
[error] dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:3194)
[error] dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:1014)
[error] dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1052)
[error] dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:327)
[error] dotty.tools.dotc.inlines.Inliner$InlineTyper.typedApply(Inliner.scala:812)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2920)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error] dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error] dotty.tools.dotc.typer.Typer.typedInlined(Typer.scala:1917)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2942)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error] dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3170)
[error] dotty.tools.dotc.inlines.Inliner$InlineTyper.typedApply(Inliner.scala:817)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2920)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error] dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3047)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error] dotty.tools.dotc.inlines.Inliner.inlined(Inliner.scala:671)
[error] dotty.tools.dotc.inlines.Inlines$InlineCall.expand(Inlines.scala:451)
[error] dotty.tools.dotc.inlines.Inlines$.inlineCall(Inlines.scala:148)
[error] dotty.tools.dotc.typer.Typer.adaptNoArgsOther$1(Typer.scala:3760)
[error] dotty.tools.dotc.typer.Typer.adaptNoArgs$1(Typer.scala:3862)
[error] dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:4082)
[error] dotty.tools.dotc.typer.Typer.adapt(Typer.scala:3431)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3170)
[error] dotty.tools.dotc.typer.Typer.typedValDef(Typer.scala:2295)
[error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2893)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2982)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3076)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3126)
[error] dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1060)
[error] dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1064)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2928)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3170)
[error] dotty.tools.dotc.typer.Namer.typedAheadExpr$$anonfun$1(Namer.scala:1619)
[error] dotty.tools.dotc.typer.Namer.typedAhead(Namer.scala:1609)
[error] dotty.tools.dotc.typer.Namer.typedAheadExpr(Namer.scala:1619)
[error] dotty.tools.dotc.typer.Namer.typedAheadRhs$1$$anonfun$1(Namer.scala:1871)
[error] dotty.tools.dotc.inlines.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:249)
[error] dotty.tools.dotc.typer.Namer.typedAheadRhs$1(Namer.scala:1871)
[error] dotty.tools.dotc.typer.Namer.rhsType$1(Namer.scala:1879)
[error] dotty.tools.dotc.typer.Namer.cookedRhsType$1(Namer.scala:1897)
[error] dotty.tools.dotc.typer.Namer.lhsType$1(Namer.scala:1898)
[error] dotty.tools.dotc.typer.Namer.inferredResultType(Namer.scala:1909)
[error] dotty.tools.dotc.typer.Namer.inferredType$1(Namer.scala:1658)
[error] dotty.tools.dotc.typer.Namer.valOrDefDefSig(Namer.scala:1665)
[error] dotty.tools.dotc.typer.Namer.defDefSig(Namer.scala:1754)
[error] dotty.tools.dotc.typer.Namer$Completer.typeSig(Namer.scala:787)
[error] dotty.tools.dotc.typer.Namer$Completer.completeInCreationContext(Namer.scala:922)
[error] dotty.tools.dotc.typer.Namer$Completer.complete(Namer.scala:810)
[error] dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:173)
[error] dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:187)
[error] dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:189)
[error] dotty.tools.dotc.core.SymDenotations$SymDenotation.ensureCompleted(SymDenotations.scala:380)
[error] dotty.tools.dotc.typer.Typer.retrieveSym(Typer.scala:2861)
[error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2886)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2982)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3076)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3126)
[error] dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:2556)
[error] dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:2908)
[error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2912)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2982)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3076)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3126)
[error] dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:2687)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2953)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3170)
[error] dotty.tools.dotc.typer.TyperPhase.typeCheck$$anonfun$1(TyperPhase.scala:44)
[error] dotty.tools.dotc.typer.TyperPhase.typeCheck$$anonfun$adapted$1(TyperPhase.scala:54)
[error] scala.Function0.apply$mcV$sp(Function0.scala:42)
[error] dotty.tools.dotc.core.Phases$Phase.monitor(Phases.scala:429)
[error] dotty.tools.dotc.typer.TyperPhase.typeCheck(TyperPhase.scala:54)
[error] dotty.tools.dotc.typer.TyperPhase.runOn$$anonfun$3(TyperPhase.scala:88)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] scala.collection.immutable.List.foreach(List.scala:333)
[error] dotty.tools.dotc.typer.TyperPhase.runOn(TyperPhase.scala:88)
[error] dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:233)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1321)
[error] dotty.tools.dotc.Run.runPhases$1(Run.scala:244)
[error] dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:252)
[error] dotty.tools.dotc.Run.compileUnits$$anonfun$adapted$1(Run.scala:261)
[error] dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:68)
[error] dotty.tools.dotc.Run.compileUnits(Run.scala:261)
[error] dotty.tools.dotc.Run.compileUnits(Run.scala:191)
[error] dotty.tools.dotc.Driver.finish(Driver.scala:56)
[error] dotty.tools.dotc.Driver.doCompile(Driver.scala:36)
[error] dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:88)
[error] dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
[error] sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:193)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:248)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:183)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:163)
[error] sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:163)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:211)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:534)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:534)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:179)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:177)
[error] sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:463)
[error] sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error] sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:263)
[error] sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:418)
[error] sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:506)
[error] sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:405)
[error] sbt.internal.inc.Incremental$.apply(Incremental.scala:171)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:534)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:488)
[error] sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:425)
[error] sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error] sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2363)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2313)
[error] sbt.internal.server.BspCompileTask$.$anonfun$compute$1(BspCompileTask.scala:30)
[error] sbt.internal.io.Retry$.apply(Retry.scala:46)
[error] sbt.internal.io.Retry$.apply(Retry.scala:28)
[error] sbt.internal.io.Retry$.apply(Retry.scala:23)
[error] sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:30)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2311)
[error] scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] sbt.Execute.work(Execute.scala:291)
[error] sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[error] java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[error] java.base/java.lang.Thread.run(Thread.java:833)
[error]            
[error] java.lang.AssertionError: assertion failed: position not set for new scala.annotation.internal.SourceFile("base/src/main/scala/demo/Base.scala") # -1 of class dotty.tools.dotc.ast.Trees$Apply in <no file>
[error]         at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
[error]         at dotty.tools.dotc.typer.Typer$.assertPositioned(Typer.scala:71)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3045)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error]         at dotty.tools.dotc.typer.Typer.$anonfun$36(Typer.scala:1898)
[error]         at dotty.tools.dotc.core.Decorators$ListDecorator$.loop$1(Decorators.scala:87)
[error]         at dotty.tools.dotc.core.Decorators$ListDecorator$.mapconserve$extension(Decorators.scala:103)
[error]         at dotty.tools.dotc.typer.Typer.typedSeqLiteral(Typer.scala:1898)
[error]         at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2941)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error]         at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error]         at dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error]         at dotty.tools.dotc.typer.Typer.typedInlined(Typer.scala:1917)
[error]         at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2942)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error]         at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error]         at dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error]         at dotty.tools.dotc.typer.Typer.$anonfun$36(Typer.scala:1898)
[error]         at dotty.tools.dotc.core.Decorators$ListDecorator$.loop$1(Decorators.scala:87)
[error]         at dotty.tools.dotc.core.Decorators$ListDecorator$.mapconserve$extension(Decorators.scala:103)
[error]         at dotty.tools.dotc.typer.Typer.typedSeqLiteral(Typer.scala:1898)
[error]         at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2941)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error]         at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error]         at dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error]         at dotty.tools.dotc.typer.ReTyper.typedTyped(ReTyper.scala:64)
[error]         at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2925)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error]         at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error]         at dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error]         at dotty.tools.dotc.typer.ProtoTypes$FunProto.$anonfun$7(ProtoTypes.scala:483)
[error]         at dotty.tools.dotc.typer.ProtoTypes$FunProto.cacheTypedArg(ProtoTypes.scala:406)
[error]         at dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:484)
[error]         at dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:866)
[error]         at dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:866)
[error]         at dotty.tools.dotc.typer.Applications$Application.addTyped$1(Applications.scala:557)
[error]         at dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:603)
[error]         at dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:460)
[error]         at dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:748)
[error]         at dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:865)
[error]         at dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1077)
[error]         at dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:327)
[error]         at dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:121)
[error]         at dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:923)
[error]         at dotty.tools.dotc.typer.Applications.realApply$1$$anonfun$3(Applications.scala:1003)
[error]         at dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:3194)
[error]         at dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:1014)
[error]         at dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1052)
[error]         at dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:327)
[error]         at dotty.tools.dotc.inlines.Inliner$InlineTyper.typedApply(Inliner.scala:812)
[error]         at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2920)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error]         at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error]         at dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error]         at dotty.tools.dotc.typer.Typer.typedInlined(Typer.scala:1917)
[error]         at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2942)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error]         at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error]         at dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error]         at dotty.tools.dotc.typer.Typer.typedInlined(Typer.scala:1917)
[error]         at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2942)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error]         at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error]         at dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error]         at dotty.tools.dotc.typer.ProtoTypes$FunProto.$anonfun$7(ProtoTypes.scala:483)
[error]         at dotty.tools.dotc.typer.ProtoTypes$FunProto.cacheTypedArg(ProtoTypes.scala:406)
[error]         at dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:484)
[error]         at dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:866)
[error]         at dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:866)
[error]         at dotty.tools.dotc.typer.Applications$Application.addTyped$1(Applications.scala:557)
[error]         at dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:622)
[error]         at dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:460)
[error]         at dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:748)
[error]         at dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:865)
[error]         at dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1077)
[error]         at dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:327)
[error]         at dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:121)
[error]         at dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:923)
[error]         at dotty.tools.dotc.typer.Applications.realApply$1$$anonfun$3(Applications.scala:1003)
[error]         at dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:3194)
[error]         at dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:1014)
[error]         at dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1052)
[error]         at dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:327)
[error]         at dotty.tools.dotc.inlines.Inliner$InlineTyper.typedApply(Inliner.scala:812)
[error]         at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2920)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error]         at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error]         at dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error]         at dotty.tools.dotc.typer.Typer.typedInlined(Typer.scala:1917)
[error]         at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2942)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error]         at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error]         at dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error]         at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3170)
[error]         at dotty.tools.dotc.inlines.Inliner$InlineTyper.typedApply(Inliner.scala:817)
[error]         at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2920)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error]         at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:126)
[error]         at dotty.tools.dotc.inlines.Inliner$InlineTyper.typedUnadapted(Inliner.scala:900)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3047)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error]         at dotty.tools.dotc.inlines.Inliner.inlined(Inliner.scala:671)
[error]         at dotty.tools.dotc.inlines.Inlines$InlineCall.expand(Inlines.scala:451)
[error]         at dotty.tools.dotc.inlines.Inlines$.inlineCall(Inlines.scala:148)
[error]         at dotty.tools.dotc.typer.Typer.adaptNoArgsOther$1(Typer.scala:3760)
[error]         at dotty.tools.dotc.typer.Typer.adaptNoArgs$1(Typer.scala:3862)
[error]         at dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:4082)
[error]         at dotty.tools.dotc.typer.Typer.adapt(Typer.scala:3431)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error]         at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3170)
[error]         at dotty.tools.dotc.typer.Typer.typedValDef(Typer.scala:2295)
[error]         at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2893)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2982)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error]         at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3076)
[error]         at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3126)
[error]         at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1060)
[error]         at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1064)
[error]         at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2928)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error]         at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3170)
[error]         at dotty.tools.dotc.typer.Namer.typedAheadExpr$$anonfun$1(Namer.scala:1619)
[error]         at dotty.tools.dotc.typer.Namer.typedAhead(Namer.scala:1609)
[error]         at dotty.tools.dotc.typer.Namer.typedAheadExpr(Namer.scala:1619)
[error]         at dotty.tools.dotc.typer.Namer.typedAheadRhs$1$$anonfun$1(Namer.scala:1871)
[error]         at dotty.tools.dotc.inlines.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:249)
[error]         at dotty.tools.dotc.typer.Namer.typedAheadRhs$1(Namer.scala:1871)
[error]         at dotty.tools.dotc.typer.Namer.rhsType$1(Namer.scala:1879)
[error]         at dotty.tools.dotc.typer.Namer.cookedRhsType$1(Namer.scala:1897)
[error]         at dotty.tools.dotc.typer.Namer.lhsType$1(Namer.scala:1898)
[error]         at dotty.tools.dotc.typer.Namer.inferredResultType(Namer.scala:1909)
[error]         at dotty.tools.dotc.typer.Namer.inferredType$1(Namer.scala:1658)
[error]         at dotty.tools.dotc.typer.Namer.valOrDefDefSig(Namer.scala:1665)
[error]         at dotty.tools.dotc.typer.Namer.defDefSig(Namer.scala:1754)
[error]         at dotty.tools.dotc.typer.Namer$Completer.typeSig(Namer.scala:787)
[error]         at dotty.tools.dotc.typer.Namer$Completer.completeInCreationContext(Namer.scala:922)
[error]         at dotty.tools.dotc.typer.Namer$Completer.complete(Namer.scala:810)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:173)
[error]         at dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:187)
[error]         at dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:189)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.ensureCompleted(SymDenotations.scala:380)
[error]         at dotty.tools.dotc.typer.Typer.retrieveSym(Typer.scala:2861)
[error]         at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2886)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2982)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error]         at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3076)
[error]         at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3126)
[error]         at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:2556)
[error]         at dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:2908)
[error]         at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2912)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2982)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error]         at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3076)
[error]         at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3126)
[error]         at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:2687)
[error]         at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2953)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2983)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3050)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3054)
[error]         at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3170)
[error]         at dotty.tools.dotc.typer.TyperPhase.typeCheck$$anonfun$1(TyperPhase.scala:44)
[error]         at dotty.tools.dotc.typer.TyperPhase.typeCheck$$anonfun$adapted$1(TyperPhase.scala:54)
[error]         at scala.Function0.apply$mcV$sp(Function0.scala:42)
[error]         at dotty.tools.dotc.core.Phases$Phase.monitor(Phases.scala:429)
[error]         at dotty.tools.dotc.typer.TyperPhase.typeCheck(TyperPhase.scala:54)
[error]         at dotty.tools.dotc.typer.TyperPhase.runOn$$anonfun$3(TyperPhase.scala:88)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error]         at scala.collection.immutable.List.foreach(List.scala:333)
[error]         at dotty.tools.dotc.typer.TyperPhase.runOn(TyperPhase.scala:88)
[error]         at dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:233)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error]         at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1321)
[error]         at dotty.tools.dotc.Run.runPhases$1(Run.scala:244)
[error]         at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:252)
[error]         at dotty.tools.dotc.Run.compileUnits$$anonfun$adapted$1(Run.scala:261)
[error]         at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:68)
[error]         at dotty.tools.dotc.Run.compileUnits(Run.scala:261)
[error]         at dotty.tools.dotc.Run.compileUnits(Run.scala:191)
[error]         at dotty.tools.dotc.Driver.finish(Driver.scala:56)
[error]         at dotty.tools.dotc.Driver.doCompile(Driver.scala:36)
[error]         at dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:88)
[error]         at dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
[error]         at sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
[error]         at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:193)
[error]         at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error]         at sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:248)
[error]         at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:183)
[error]         at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:163)
[error]         at sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error]         at sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:163)
[error]         at sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:211)
[error]         at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:534)
[error]         at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:534)
[error]         at sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:179)
[error]         at sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:177)
[error]         at sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:463)
[error]         at sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error]         at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error]         at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error]         at sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:263)
[error]         at sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:418)
[error]         at sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:506)
[error]         at sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:405)
[error]         at sbt.internal.inc.Incremental$.apply(Incremental.scala:171)
[error]         at sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:534)
[error]         at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:488)
[error]         at sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error]         at sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:425)
[error]         at sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error]         at sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2363)
[error]         at sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2313)
[error]         at sbt.internal.server.BspCompileTask$.$anonfun$compute$1(BspCompileTask.scala:30)
[error]         at sbt.internal.io.Retry$.apply(Retry.scala:46)
[error]         at sbt.internal.io.Retry$.apply(Retry.scala:28)
[error]         at sbt.internal.io.Retry$.apply(Retry.scala:23)
[error]         at sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:30)
[error]         at sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2311)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error]         at sbt.Execute.work(Execute.scala:291)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[error]         at java.base/java.lang.Thread.run(Thread.java:833)
[error] (dep / Compile / compileIncremental) java.lang.AssertionError: assertion failed: position not set for new scala.annotation.internal.SourceFile("base/src/main/scala/demo/Base.scala") # -1 of class dotty.tools.dotc.ast.Trees$Apply in <no file>
```
</details>

--- a/compiler/src/dotty/tools/dotc/ast/Desugar.scala
+++ b/compiler/src/dotty/tools/dotc/ast/Desugar.scala
@@ -175,7 +175,7 @@ object desugar {
       case Into(tpt1) =>
         mods1 = vdef.mods.withAddedAnnotation(
           TypedSplice(
-            Annotation(defn.AllowConversionsAnnot).tree.withSpan(tpt.span.startPos)))
+            Annotation(defn.AllowConversionsAnnot, tpt.span.startPos).tree))
         tpt1
       case ByNameTypeTree(tpt1) =>
         cpy.ByNameTypeTree(tpt)(dropInto(tpt1))

--- a/compiler/src/dotty/tools/dotc/core/Annotations.scala
+++ b/compiler/src/dotty/tools/dotc/core/Annotations.scala
@@ -194,36 +194,18 @@ object Annotations {
   object Annotation {
 
     def apply(tree: Tree): ConcreteAnnotation = ConcreteAnnotation(tree)
-
-    def apply(cls: ClassSymbol)(using Context): Annotation =
-      apply(cls, Nil)
     
     def apply(cls: ClassSymbol, span: Span)(using Context): Annotation =
       apply(cls, Nil, span)
 
-    def apply(cls: ClassSymbol, arg: Tree)(using Context): Annotation =
-      apply(cls, arg :: Nil)
-    
     def apply(cls: ClassSymbol, arg: Tree, span: Span)(using Context): Annotation =
       apply(cls, arg :: Nil, span)
 
-    def apply(cls: ClassSymbol, arg1: Tree, arg2: Tree)(using Context): Annotation =
-      apply(cls, arg1 :: arg2 :: Nil)
-
-    def apply(cls: ClassSymbol, args: List[Tree])(using Context): Annotation =
-      apply(cls.typeRef, args)
-    
     def apply(cls: ClassSymbol, args: List[Tree], span: Span)(using Context): Annotation =
       apply(cls.typeRef, args, span)
 
-    def apply(atp: Type, arg: Tree)(using Context): Annotation =
-      apply(atp, arg :: Nil)
-
-    def apply(atp: Type, arg1: Tree, arg2: Tree)(using Context): Annotation =
-      apply(atp, arg1 :: arg2 :: Nil)
-
-    def apply(atp: Type, args: List[Tree])(using Context): Annotation =
-      apply(New(atp, args))
+    def apply(atp: Type, arg: Tree, span: Span)(using Context): Annotation =
+      apply(atp, arg :: Nil, span)
     
     def apply(atp: Type, args: List[Tree], span: Span)(using Context): Annotation =
       apply(New(atp, args).withSpan(span))
@@ -271,7 +253,7 @@ object Annotations {
 
   def ThrowsAnnotation(cls: ClassSymbol)(using Context): Annotation = {
     val tref = cls.typeRef
-    Annotation(defn.ThrowsAnnot.typeRef.appliedTo(tref), Ident(tref))
+    Annotation(defn.ThrowsAnnot.typeRef.appliedTo(tref), Ident(tref), cls.span)
   }
 
   /** Extracts the type of the thrown exception from an annotation.

--- a/compiler/src/dotty/tools/dotc/core/Annotations.scala
+++ b/compiler/src/dotty/tools/dotc/core/Annotations.scala
@@ -197,15 +197,24 @@ object Annotations {
 
     def apply(cls: ClassSymbol)(using Context): Annotation =
       apply(cls, Nil)
+    
+    def apply(cls: ClassSymbol, span: Span)(using Context): Annotation =
+      apply(cls, Nil, span)
 
     def apply(cls: ClassSymbol, arg: Tree)(using Context): Annotation =
       apply(cls, arg :: Nil)
+    
+    def apply(cls: ClassSymbol, arg: Tree, span: Span)(using Context): Annotation =
+      apply(cls, arg :: Nil, span)
 
     def apply(cls: ClassSymbol, arg1: Tree, arg2: Tree)(using Context): Annotation =
       apply(cls, arg1 :: arg2 :: Nil)
 
     def apply(cls: ClassSymbol, args: List[Tree])(using Context): Annotation =
       apply(cls.typeRef, args)
+    
+    def apply(cls: ClassSymbol, args: List[Tree], span: Span)(using Context): Annotation =
+      apply(cls.typeRef, args, span)
 
     def apply(atp: Type, arg: Tree)(using Context): Annotation =
       apply(atp, arg :: Nil)
@@ -215,6 +224,9 @@ object Annotations {
 
     def apply(atp: Type, args: List[Tree])(using Context): Annotation =
       apply(New(atp, args))
+    
+    def apply(atp: Type, args: List[Tree], span: Span)(using Context): Annotation =
+      apply(New(atp, args).withSpan(span))
 
     /** Create an annotation where the tree is computed lazily. */
     def deferred(sym: Symbol)(treeFn: Context ?=> Tree): Annotation =
@@ -251,8 +263,8 @@ object Annotations {
         else None
     }
 
-    def makeSourceFile(path: String)(using Context): Annotation =
-      apply(defn.SourceFileAnnot, Literal(Constant(path)))
+    def makeSourceFile(path: String, span: Span)(using Context): Annotation =
+      apply(defn.SourceFileAnnot, Literal(Constant(path)), span)
   }
 
   @sharable val EmptyAnnotation = Annotation(EmptyTree)

--- a/compiler/src/dotty/tools/dotc/core/SymDenotations.scala
+++ b/compiler/src/dotty/tools/dotc/core/SymDenotations.scala
@@ -283,7 +283,7 @@ object SymDenotations {
 
     /** Add the given annotation without parameters to the annotations of this denotation */
     final def addAnnotation(cls: ClassSymbol)(using Context): Unit =
-      addAnnotation(Annotation(cls))
+      addAnnotation(Annotation(cls, symbol.span))
 
     /** Remove annotation with given class from this denotation */
     final def removeAnnotation(cls: Symbol)(using Context): Unit =

--- a/compiler/src/dotty/tools/dotc/core/Types.scala
+++ b/compiler/src/dotty/tools/dotc/core/Types.scala
@@ -3965,9 +3965,9 @@ object Types {
      *   - wrap types of parameters that have an @allowConversions annotation with Into[_]
      */
     def fromSymbols(params: List[Symbol], resultType: Type)(using Context): MethodType =
-      def addAnnotation(tp: Type, cls: ClassSymbol): Type = tp match
-        case ExprType(resType) => ExprType(addAnnotation(resType, cls))
-        case _ => AnnotatedType(tp, Annotation(cls))
+      def addAnnotation(tp: Type, cls: ClassSymbol, param: Symbol): Type = tp match
+        case ExprType(resType) => ExprType(addAnnotation(resType, cls, param))
+        case _ => AnnotatedType(tp, Annotation(cls, param.span))
 
       def wrapConvertible(tp: Type) =
         AppliedType(defn.IntoType.typeRef, tp :: Nil)
@@ -3992,9 +3992,9 @@ object Types {
       def paramInfo(param: Symbol) =
         var paramType = param.info.annotatedToRepeated
         if param.is(Inline) then
-          paramType = addAnnotation(paramType, defn.InlineParamAnnot)
+          paramType = addAnnotation(paramType, defn.InlineParamAnnot, param)
         if param.is(Erased) then
-          paramType = addAnnotation(paramType, defn.ErasedParamAnnot)
+          paramType = addAnnotation(paramType, defn.ErasedParamAnnot, param)
         if param.hasAnnotation(defn.AllowConversionsAnnot) then
           paramType = addInto(paramType)
         paramType

--- a/compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala
+++ b/compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala
@@ -764,7 +764,7 @@ class ClassfileParser(
               res.namedParams += (i -> name.name)
 
         case tpnme.AnnotationDefaultATTR =>
-          sym.addAnnotation(Annotation(defn.AnnotationDefaultAnnot, Nil))
+          sym.addAnnotation(Annotation(defn.AnnotationDefaultAnnot, Nil, sym.span))
 
         // Java annotations on classes / methods / fields with RetentionPolicy.RUNTIME
         case tpnme.RuntimeVisibleAnnotationATTR

--- a/compiler/src/dotty/tools/dotc/transform/CompleteJavaEnums.scala
+++ b/compiler/src/dotty/tools/dotc/transform/CompleteJavaEnums.scala
@@ -110,7 +110,7 @@ class CompleteJavaEnums extends MiniPhase with InfoTransformer { thisPhase =>
     yield {
       def forwarderSym(flags: FlagSet, info: Type): Symbol { type ThisName = TermName } =
         val sym = newSymbol(clazz, enumValue.name.asTermName, flags, info)
-        sym.addAnnotation(Annotations.Annotation(defn.ScalaStaticAnnot))
+        sym.addAnnotation(Annotations.Annotation(defn.ScalaStaticAnnot, sym.span))
         sym
       val body = moduleRef.select(enumValue)
       if ctx.settings.scalajs.value then

--- a/compiler/src/dotty/tools/dotc/transform/ContextFunctionResults.scala
+++ b/compiler/src/dotty/tools/dotc/transform/ContextFunctionResults.scala
@@ -39,7 +39,7 @@ object ContextFunctionResults:
     val count = contextResultCount(mdef.rhs, mdef.tpt.tpe)
 
     if Config.flattenContextFunctionResults && count != 0 && !disabled then
-      val countAnnot = Annotation(defn.ContextResultCountAnnot, Literal(Constant(count)))
+      val countAnnot = Annotation(defn.ContextResultCountAnnot, Literal(Constant(count)), mdef.symbol.span)
       mdef.symbol.addAnnotation(countAnnot)
   end annotateContextResults
 

--- a/compiler/src/dotty/tools/dotc/transform/LazyVals.scala
+++ b/compiler/src/dotty/tools/dotc/transform/LazyVals.scala
@@ -112,7 +112,7 @@ class LazyVals extends MiniPhase with IdentityDenotTransformer {
     appendOffsetDefs.get(cls) match {
       case None => template
       case Some(data) =>
-        data.defs.foreach(_.symbol.addAnnotation(Annotation(defn.ScalaStaticAnnot)))
+        data.defs.foreach(defin => defin.symbol.addAnnotation(Annotation(defn.ScalaStaticAnnot, defin.symbol.span)))
         cpy.Template(template)(body = addInFront(data.defs, template.body))
     }
   }
@@ -464,7 +464,7 @@ class LazyVals extends MiniPhase with IdentityDenotTransformer {
     def offsetName(id: Int) = s"${StdNames.nme.LAZY_FIELD_OFFSET}${if (x.symbol.owner.is(Module)) "_m_" else ""}$id".toTermName
     val containerName = LazyLocalName.fresh(x.name.asTermName)
     val containerSymbol = newSymbol(claz, containerName, x.symbol.flags &~ containerFlagsMask | containerFlags | Private, defn.ObjectType, coord = x.symbol.coord).enteredAfter(this)
-    containerSymbol.addAnnotation(Annotation(defn.VolatileAnnot)) // private @volatile var _x: AnyRef
+    containerSymbol.addAnnotation(Annotation(defn.VolatileAnnot, containerSymbol.span)) // private @volatile var _x: AnyRef
     containerSymbol.addAnnotations(x.symbol.annotations) // pass annotations from original definition
     val stat = x.symbol.isStatic
     if stat then
@@ -482,7 +482,7 @@ class LazyVals extends MiniPhase with IdentityDenotTransformer {
         newSymbol(claz, offsetName(info.defs.size), Synthetic, defn.LongType).enteredAfter(this)
       case None =>
         newSymbol(claz, offsetName(0), Synthetic, defn.LongType).enteredAfter(this)
-    offsetSymbol.nn.addAnnotation(Annotation(defn.ScalaStaticAnnot))
+    offsetSymbol.nn.addAnnotation(Annotation(defn.ScalaStaticAnnot, offsetSymbol.nn.span))
     val fieldTree = thizClass.select(lazyNme.RLazyVals.getDeclaredField).appliedTo(Literal(Constant(containerName.mangledString)))
     val offsetTree = ValDef(offsetSymbol.nn, getOffset.appliedTo(fieldTree))
     val offsetInfo = appendOffsetDefs.getOrElseUpdate(claz, new OffsetInfo(Nil))
@@ -625,7 +625,7 @@ class LazyVals extends MiniPhase with IdentityDenotTransformer {
              .symbol.asTerm
         else { // need to create a new flag
           offsetSymbol = newSymbol(claz, offsetById, Synthetic, defn.LongType).enteredAfter(this)
-          offsetSymbol.nn.addAnnotation(Annotation(defn.ScalaStaticAnnot))
+          offsetSymbol.nn.addAnnotation(Annotation(defn.ScalaStaticAnnot, offsetSymbol.nn.span))
           val flagName = LazyBitMapName.fresh(id.toString.toTermName)
           val flagSymbol = newSymbol(claz, flagName, containerFlags, defn.LongType).enteredAfter(this)
           flag = ValDef(flagSymbol, Literal(Constant(0L)))
@@ -636,7 +636,7 @@ class LazyVals extends MiniPhase with IdentityDenotTransformer {
 
       case None =>
         offsetSymbol = newSymbol(claz, offsetName(0), Synthetic, defn.LongType).enteredAfter(this)
-        offsetSymbol.nn.addAnnotation(Annotation(defn.ScalaStaticAnnot))
+        offsetSymbol.nn.addAnnotation(Annotation(defn.ScalaStaticAnnot, offsetSymbol.nn.span))
         val flagName = LazyBitMapName.fresh("0".toTermName)
         val flagSymbol = newSymbol(claz, flagName, containerFlags, defn.LongType).enteredAfter(this)
         flag = ValDef(flagSymbol, Literal(Constant(0L)))

--- a/compiler/src/dotty/tools/dotc/transform/MoveStatics.scala
+++ b/compiler/src/dotty/tools/dotc/transform/MoveStatics.scala
@@ -46,7 +46,7 @@ class MoveStatics extends MiniPhase with SymTransformer {
           if (staticFields.nonEmpty) {
             /* do NOT put Flags.JavaStatic here. It breaks .enclosingClass */
             val staticCostructor = newSymbol(orig.symbol, nme.STATIC_CONSTRUCTOR, Flags.Synthetic | Flags.Method | Flags.Private, MethodType(Nil, defn.UnitType))
-            staticCostructor.addAnnotation(Annotation(defn.ScalaStaticAnnot))
+            staticCostructor.addAnnotation(Annotation(defn.ScalaStaticAnnot, staticCostructor.span))
             staticCostructor.entered
 
             val staticAssigns = staticFields.map(x => Assign(ref(x.symbol), x.rhs.changeOwner(x.symbol, staticCostructor)))

--- a/compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala
+++ b/compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala
@@ -297,7 +297,7 @@ object PCPCheckAndHeal {
         flags = Synthetic,
         info = TypeAlias(splicedTree.tpe.select(tpnme.Underlying)),
         coord = span).asType
-      local.addAnnotation(Annotation(defn.QuotedRuntime_SplicedTypeAnnot))
+      local.addAnnotation(Annotation(defn.QuotedRuntime_SplicedTypeAnnot, span))
       ctx.typeAssigner.assignType(untpd.TypeDef(local.name, alias), local)
     }
 

--- a/compiler/src/dotty/tools/dotc/transform/PostTyper.scala
+++ b/compiler/src/dotty/tools/dotc/transform/PostTyper.scala
@@ -395,9 +395,9 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
               if ctx.compilationUnit.source.exists && sym != defn.SourceFileAnnot then
                 val reference = ctx.settings.sourceroot.value
                 val relativePath = util.SourceFile.relativePath(ctx.compilationUnit.source, reference)
-                sym.addAnnotation(Annotation.makeSourceFile(relativePath))
+                sym.addAnnotation(Annotation.makeSourceFile(relativePath, tree.span))
               if Feature.pureFunsEnabled && sym != defn.WithPureFunsAnnot then
-                sym.addAnnotation(Annotation(defn.WithPureFunsAnnot))
+                sym.addAnnotation(Annotation(defn.WithPureFunsAnnot, tree.span))
           else
             if !sym.is(Param) && !sym.owner.isOneOf(AbstractOrTrait) then
               Checking.checkGoodBounds(tree.symbol)
@@ -499,8 +499,8 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
 
     private def annotateExperimental(sym: Symbol)(using Context): Unit =
       if sym.is(Module) && sym.companionClass.hasAnnotation(defn.ExperimentalAnnot) then
-        sym.addAnnotation(defn.ExperimentalAnnot)
-        sym.companionModule.addAnnotation(defn.ExperimentalAnnot)
+        sym.addAnnotation(Annotation(defn.ExperimentalAnnot, sym.span))
+        sym.companionModule.addAnnotation(Annotation(defn.ExperimentalAnnot, sym.span))
 
   }
 }

--- a/compiler/src/dotty/tools/dotc/transform/RepeatableAnnotations.scala
+++ b/compiler/src/dotty/tools/dotc/transform/RepeatableAnnotations.scala
@@ -45,7 +45,7 @@ class RepeatableAnnotations extends MiniPhase:
               Nil
             else
               val aggregated = JavaSeqLiteral(anns.map(_.tree).toList, TypeTree(sym.typeRef))
-              Annotation(containerTpe, NamedArg("value".toTermName, aggregated)) :: Nil
+              Annotation(containerTpe, NamedArg("value".toTermName, aggregated), sym.span) :: Nil
           case _ =>
             val pos = anns.head.tree.srcPos
             report.error("Not repeatable annotation repeated", pos)

--- a/compiler/src/dotty/tools/dotc/transform/SymUtils.scala
+++ b/compiler/src/dotty/tools/dotc/transform/SymUtils.scala
@@ -381,7 +381,7 @@ object SymUtils:
       if original.hasAnnotation(defn.TargetNameAnnot) then
         self.addAnnotation(
           Annotation(defn.TargetNameAnnot,
-            Literal(Constant(nameFn(original.targetName).toString)).withSpan(original.span)))
+            Literal(Constant(nameFn(original.targetName).toString)).withSpan(original.span), original.span))
 
     /** The return type as seen from the body of this definition. It is
      *  computed from the symbol's type by replacing param refs by param symbols.

--- a/compiler/src/dotty/tools/dotc/transform/SyntheticMembers.scala
+++ b/compiler/src/dotty/tools/dotc/transform/SyntheticMembers.scala
@@ -13,6 +13,7 @@ import ast.untpd
 import ValueClasses.isDerivedValueClass
 import SymUtils._
 import util.Property
+import util.Spans.Span
 import config.Printers.derive
 import NullOpsDecorator._
 
@@ -155,7 +156,7 @@ class SyntheticMembers(thisPhase: DenotTransformer) {
         case nme.hashCode_ => chooseHashcode
         case nme.toString_ => toStringBody(vrefss)
         case nme.equals_ => equalsBody(vrefss.head.head)
-        case nme.canEqual_ => canEqualBody(vrefss.head.head)
+        case nme.canEqual_ => canEqualBody(vrefss.head.head, synthetic.span)
         case nme.ordinal => ordinalRef
         case nme.productArity => Literal(Constant(accessors.length))
         case nme.productPrefix if isEnumValue => nameRef
@@ -260,7 +261,7 @@ class SyntheticMembers(thisPhase: DenotTransformer) {
     def equalsBody(that: Tree)(using Context): Tree = {
       val thatAsClazz = newSymbol(ctx.owner, nme.x_0, SyntheticCase, clazzType, coord = ctx.owner.span) // x$0
       def wildcardAscription(tp: Type) = Typed(Underscore(tp), TypeTree(tp))
-      val pattern = Bind(thatAsClazz, wildcardAscription(AnnotatedType(clazzType, Annotation(defn.UncheckedAnnot)))) // x$0 @ (_: C @unchecked)
+      val pattern = Bind(thatAsClazz, wildcardAscription(AnnotatedType(clazzType, Annotation(defn.UncheckedAnnot, thatAsClazz.span)))) // x$0 @ (_: C @unchecked)
       // compare primitive fields first, slow equality checks of non-primitive fields can be skipped when primitives differ
       val sortedAccessors = accessors.sortBy(accessor => if (accessor.info.typeSymbol.isPrimitiveValueClass) 0 else 1)
       val comparisons = sortedAccessors.map { accessor =>
@@ -390,7 +391,7 @@ class SyntheticMembers(thisPhase: DenotTransformer) {
      *
      *  `@unchecked` is needed for parametric case classes.
      */
-    def canEqualBody(that: Tree): Tree = that.isInstance(AnnotatedType(clazzType, Annotation(defn.UncheckedAnnot)))
+    def canEqualBody(that: Tree, span: Span): Tree = that.isInstance(AnnotatedType(clazzType, Annotation(defn.UncheckedAnnot, span)))
 
     symbolsToSynthesize.flatMap(syntheticDefIfMissing)
   }

--- a/compiler/src/dotty/tools/dotc/typer/Namer.scala
+++ b/compiler/src/dotty/tools/dotc/typer/Namer.scala
@@ -461,7 +461,7 @@ class Namer { typer: Typer =>
         val isProvisional = parents.exists(!_.baseType(defn.AnyClass).exists)
         if isProvisional then
           typr.println(i"provisional superclass $first for $cls")
-          first = AnnotatedType(first, Annotation(defn.ProvisionalSuperClassAnnot))
+          first = AnnotatedType(first, Annotation(defn.ProvisionalSuperClassAnnot, cls.span))
         checkFeasibleParent(first, cls.srcPos, i" in inferred superclass $first") :: parents
   end ensureFirstIsClass
 
@@ -1883,7 +1883,7 @@ class Namer { typer: Typer =>
         // larger choice of overrides (see `default-getter.scala`).
         // For justification on the use of `@uncheckedVariance`, see
         // `default-getter-variance.scala`.
-        AnnotatedType(defaultTp, Annotation(defn.UncheckedVarianceAnnot))
+        AnnotatedType(defaultTp, Annotation(defn.UncheckedVarianceAnnot, sym.span))
       else
         // don't strip @uncheckedVariance annot for default getters
         TypeOps.simplify(tp.widenTermRefExpr,

--- a/compiler/src/dotty/tools/dotc/typer/Typer.scala
+++ b/compiler/src/dotty/tools/dotc/typer/Typer.scala
@@ -1663,7 +1663,7 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
 
                 // skip exhaustivity check in later phase
                 // TODO: move the check above to patternMatcher phase
-                val uncheckedTpe = AnnotatedType(sel.tpe.widen, Annotation(defn.UncheckedAnnot))
+                val uncheckedTpe = AnnotatedType(sel.tpe.widen, Annotation(defn.UncheckedAnnot, tree.selector.span))
                 tpd.cpy.Match(result)(
                   selector = tpd.Typed(sel, tpd.TypeTree(uncheckedTpe)),
                   cases = result.cases
@@ -1898,7 +1898,7 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
       Typed(res,
         TypeTree(
           AnnotatedType(res.tpe,
-            Annotation(defn.RequiresCapabilityAnnot, cap))))
+            Annotation(defn.RequiresCapabilityAnnot, cap, tree.span))))
     else res
 
   def typedSeqLiteral(tree: untpd.SeqLiteral, pt: Type)(using Context): SeqLiteral = {
@@ -4297,7 +4297,7 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
           // this is needed for -Ycheck. Without the annotation Ycheck will
           // skolemize the result type which will lead to different types before
           // and after checking. See i11955.scala.
-          AnnotatedType(conj, Annotation(defn.UncheckedStableAnnot))
+          AnnotatedType(conj, Annotation(defn.UncheckedStableAnnot, tree.symbol.span))
         else conj
       else pt
     gadts.println(i"insert GADT cast from $tree to $target")

--- a/compiler/test/dotty/tools/dotc/transform/TypeTestsCastsTest.scala
+++ b/compiler/test/dotty/tools/dotc/transform/TypeTestsCastsTest.scala
@@ -6,6 +6,8 @@ import core.*
 import Contexts.*, Decorators.*, Denotations.*, SymDenotations.*, Symbols.*, Types.*
 import Annotations.*
 
+import dotty.tools.dotc.util.Spans.Span
+
 import org.junit.Test
 import org.junit.Assert.*
 
@@ -15,7 +17,7 @@ class TypeTestsCastsTest extends DottyTest:
   @Test def orL = checkFound(List(StringType, LongType), OrType(LongType, StringType, false))
   @Test def orR = checkFound(List(LongType, StringType), OrType(StringType, LongType, false))
 
-  @Test def annot = checkFound(List(StringType, LongType), AnnotatedType(OrType(LongType, StringType, false), Annotation(defn.UncheckedAnnot)))
+  @Test def annot = checkFound(List(StringType, LongType), AnnotatedType(OrType(LongType, StringType, false), Annotation(defn.UncheckedAnnot, Span(0))))
 
   @Test def andL = checkFound(List(StringType), AndType(StringType, AnyType))
   @Test def andR = checkFound(List(StringType), AndType(AnyType, StringType))

--- a/tests/pos-macros/i16318/Macro_1.scala
+++ b/tests/pos-macros/i16318/Macro_1.scala
@@ -1,0 +1,11 @@
+import scala.quoted.*
+
+final case class Record(a: String, b: Int)
+
+transparent inline def ann[T]: List[Any] = ${ annsImpl[T] }
+
+def annsImpl[T: Type](using Quotes): Expr[List[Any]] = {
+  import quotes.reflect.*
+  val annExpr = TypeRepr.of[T].typeSymbol.annotations.head.asExpr
+  '{ List($annExpr) }
+}

--- a/tests/pos-macros/i16318/Test_2.scala
+++ b/tests/pos-macros/i16318/Test_2.scala
@@ -1,0 +1,2 @@
+def Test =
+  val a = ann[Record]

